### DISCUSSION TITLE
[Feature]: Does rspack support rendering of jsx code snippets in .vue files?

### DISCUSSION THREAD

**Discussion by shiwenna** [web-infra-dev/rspack#8494] [Q&A]
### What problem does this feature solve?

控制台报错如下:![Image](https://github.com/user-attachments/assets/578cf267-7dfe-45e3-aba2-58befbe74f63)
node:16.20.2;
vue:2.7.16
respack配置如下：
![Image](https://github.com/user-attachments/assets/af472df9-95ba-4fd9-8877-fb9dfacc5a96)

### What does the proposed API of configuration look like?

使用pluginVue2Jsx（）不支持rspack，只支持rsbuild，rspack里的应该如何支持此类代码的loader渲染，应该如何配置？

**Comment by RainedX** [comment]
you can.use babel-loader for jsx

**Comment by shiwenna** [comment]
![Image](https://github.com/user-attachments/assets/5c3574d1-1244-487e-8c2a-c8e0ec80dab8)
单独为jsx配置了babel-loader，但是好像没有起作用，依旧报错

**Reply by limuen** [reply]
{
        test: /\.[jt]sx$/,
        use: {
          loader: "babel-loader",
          options: {
            presets: [
              ["@vue/babel-preset-jsx", { compositionAPI: true }]
            ]
          }
        },

      },
      
      再从script 上写 lang='jsx'试试呢？

**Reply by shiwenna** [reply]
![image](https://github.com/user-attachments/assets/97b54b67-9ce1-49c9-8ace-28851faa7b3f)
加了的，不行

**Reply by limuen** [reply]
依赖@vue/babel-preset-jsx装了吗？

**Reply by limuen** [reply]
跟目录下的babel.文件给删了

**Reply by shiwenna** [reply]
试验过了，不行。安装依赖@vue/babel-preset-jsx了也删掉babel文件了

**Reply by limuen** [reply]
尝试修改一下style呢？

**Comment by sxy123450** [comment]
解决了吗，我也想知道怎么解决这个问题

**Reply by ununian** [reply]
<img width="344" alt="image" src="https://github.com/user-attachments/assets/ecea2937-3d2c-40c6-83f5-55989ae67b96" />
这样配置，然后写上 lang='tsx' 就行

**Reply by shiwenna** [reply]
rspack里使用pluginVue2Jsx会报错

https://github.com/rspack-contrib/rsbuild-plugin-vue2-jsx/issues/6

**Reply by ununian** [reply]
哦  我这里是 rsbuild ，可以正确运行
